From a03c4a4ea3fb42b347bada24faae1d31c7d0c90e Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Tue, 1 Jun 2021 16:56:40 +0200
Subject: [PATCH] failonerror_test: skip the test with curl-7.75.0+

libcurl does not provide the reason phrase from HTTP status line
in the error buffer any more:

    https://github.com/curl/curl/issues/6615

Fixes: https://github.com/pycurl/pycurl/issues/679
---
 tests/failonerror_test.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/failonerror_test.py b/tests/failonerror_test.py
index dc4d8cd6b..519aed821 100644
--- a/tests/failonerror_test.py
+++ b/tests/failonerror_test.py
@@ -21,6 +21,8 @@ def tearDown(self):
     # not sure what the actual min is but 7.26 is too old
     # and does not include status text, only the status code
     @util.min_libcurl(7, 38, 0)
+    # no longer supported by libcurl: https://github.com/curl/curl/issues/6615
+    @util.removed_in_libcurl(7, 75, 0)
     def test_failonerror(self):
         self.curl.setopt(pycurl.URL, 'http://%s:8380/status/403' % localhost)
         sio = util.BytesIO()
@@ -41,6 +43,8 @@ def test_failonerror(self):
     # not sure what the actual min is but 7.26 is too old
     # and does not include status text, only the status code
     @util.min_libcurl(7, 38, 0)
+    # no longer supported by libcurl: https://github.com/curl/curl/issues/6615
+    @util.removed_in_libcurl(7, 75, 0)
     def test_failonerror_status_line_invalid_utf8_python2(self):
         self.curl.setopt(pycurl.URL, 'http://%s:8380/status_invalid_utf8' % localhost)
         sio = util.BytesIO()
@@ -61,6 +65,8 @@ def test_failonerror_status_line_invalid_utf8_python2(self):
     # not sure what the actual min is but 7.26 is too old
     # and does not include status text, only the status code
     @util.min_libcurl(7, 38, 0)
+    # no longer supported by libcurl: https://github.com/curl/curl/issues/6615
+    @util.removed_in_libcurl(7, 75, 0)
     def test_failonerror_status_line_invalid_utf8_python3(self):
         self.curl.setopt(pycurl.URL, 'http://%s:8380/status_invalid_utf8' % localhost)
         sio = util.BytesIO()
